### DISCUSSION TITLE
spacy.load() on python 3.12 with vscode

### DISCUSSION THREAD

**Discussion by lsmith77** [explosion/spaCy#13266] [Help: Other Questions]
I think this is technically not a bug in this repo but it will likely affect other people too.

I have also found the same behavior in this https://github.com/carpedm20/emoji/issues/280, which led to the creation of this ticket https://github.com/microsoft/debugpy/issues/1496

## How to reproduce the behaviour
Call `spacy.load()` using Python 3.12 when launching using VSCode.

## Your Environment

* Operating System: OSX
* Python Version Used: 3.12.1
* spaCy Version Used: 3.7
* Environment Information:

**Comment by svlandeg** [comment]
Thanks for the report!

So basically the issue is that `spacy.load` is much slower with Python 3.12 in VSCode? But it's an upstream issue?

**Comment by lsmith77** [comment]
Yeah that is what I suspect is the case. The issue is caused by large python files since Python 3.12 triggers an event for each line slowing the debugger down. They appear to be refactoring the VSCode to use some new 3.12 dedicated APIs.

In theory spacy could "fix" this by changing relevant files to put everything on a single line.

**Comment by svlandeg** [comment]
Right, thanks for the additional explanation!

I don't think we'd want to adjust spaCy's core files to accomodate this "feature" by VSCode at this point in time, but let me move this to the discussion forum so we can keep track of this thread and keep the discussion open to anyone.
